WcaOnRails/app/controllers/competition_tabs_controller.rb

```diff
@@ -43,12 +43,13 @@ def update
 
   def destroy
     @competition = competition_from_params
-    CompetitionTab.find_by(id: params[:id])&.destroy
+    CompetitionTab.find_by(id: params[:id], competition: @competition)&.destroy
```

Comment:
Is there a reason we aren't using `find_by!` here?

Reply:
From what I looked at, having find_by! would throw an exception, and here we are already accounting for it being potentially null with the & so I thought not having it throw in this case was closer to the original implementation

Reply:
Ah I see! In this case I'm partial to raising an exception as that's more consistent with our other implementations, and this is a case where an exception seems appropriate, but I can see the argument both ways.

Reply:
Reasonable, changes made in latest commit